plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/config/OpenLineageClientHttpTransportConfig.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openlineage.config;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class OpenLineageClientHttpTransportConfig
+{
+    private String url;
+    private String endpoint;
+    private Optional<String> apiKey = Optional.empty();
+    private Integer timeoutInMillis = 5000;
+    private Map<String, String> headers = new HashMap<>();
+    private Map<String, String> urlParams = new HashMap<>();
+
+    @NotNull
+    public String getUrl()
+    {
+        return url;
+    }
+
```

Comment:
Should we make it consistent to how Spark and Flink configs are given
https://openlineage.io/docs/client/java/configuration/#transports

e.g.
`trino.openlineage.transport.url`
`trino.openlineage.transport.endpoint`
`trino.openlineage.transport.auth.type`
`trino.openlineage.transport.auth.apiKey`

Reply:
I've adjusted naming convention, please see https://github.com/trinodb/trino/pull/21265/commits/a40786a4fedd9c739a22dd177bf4a4c4db14bbee